dpctl/_sycl_queue.pyx

```diff
@@ -356,8 +370,11 @@ cdef DPCTLSyclEventRef _memcpy_impl(
     if isinstance(dst, _Memory):
         c_dst_ptr = <void*>(<_Memory>dst).get_data_ptr()
     elif _is_buffer(dst):
-        dst_host_buf = dst
-        c_dst_ptr = <void *>&dst_host_buf[0]
+        ret_code = PyObject_GetBuffer(dst, &dst_buf_view, PyBUF_SIMPLE | PyBUF_ANY_CONTIGUOUS | PyBUF_WRITABLE)
+        if ret_code != 0: # pragma: no cover
+            raise RuntimeError("Could not access buffer")
```

Comment:
Should the source buffer be released before the exception is raised?

Reply:
I was asking about the case thar gret buffer was successful on source buffer but failed on destination buffer

Reply:
Sorry, I misunderstood earlier.

It seems unlikely that both `src` and `dst` are Python buffers (we'd essentially copy from Python buffer to Python buffer in that case), but I've still added a fix to release the source buffer in that case.

Reply:
Thanks for pushing the fix @sommerlukas . The use case of copying from host to host is also supported by `queue::memcpy`.

Reply:
Ok 👍 

This should be safe now with the latest change.